### PR TITLE
Review Domain entity & remove DomainViewBuilder

### LINKED ISSUE #148: DomainViewBuilder review
There are some oddities noted in DomainViewBuilder. Investigate.

### PR DESCRIPTION
**Pull request by andypost**
Closes #148 

- Removes `DomainEntityBuilder` because it suppose only for "viewable" entities
- Use `count()` in `preCreate()` instead of loading entities
- `status` & `uuid` properties are defined in base class so should not be duplicated
- `status` annotation should be set for config entities that allowed enable/disable- Clean-up doc-blocks to conform current standards (domain & interface now passes codereview)
- Removes `isDefaultRevision` method because it defined for content entities `\Drupal\Core\Entity\RevisionableInterface::isDefaultRevision()`

**TODO** looks `getRawPath()` & `buildUrl()` methods are defined but never used, makes sense to remove (they are not a part of interface)

### CODE REVIEW

**Comment by agentrickard** [comment]
buildURL() is used in DomainAccessField.php, probably as a shim for a case where I couldn't figure out how to generate a proper link without a route. (And to prevent, I think, the form from linking to the domain entity.)

In D7 we used similar functions to build absolute links, but they likely aren't needed now.

File: domain/src/Entity/Domain.php [hunk 1]
```diff
@@ -475,11 +456,4 @@ public function getMatchType() {
     return $this->matchType;
   }
 
- /**
-  * This is being fired for some reason if the values are shown on a
```
Comment (agentrickard) [inline domain/src/Entity/Domain.php:479]: This was originally here because (I think) of a bug in core.

**Comment by andypost** [comment]
Thanx for pointer, will dig about link generation

**Comment by agentrickard** [comment]
It's worth noting that the DomainLoader class was created because Drupal 8.0 wasn't quite stable from an entity API pov. There were methods that I wanted that were missing, and I wanted something stable to develop against.

It is possible that we might be able to remove the Loader (and the DomainCreator) now that core is much more advanced.

I think we're fine keeping both but wanted to note that. I have no real interest in rewriting to remove those classes.

File: domain/src/Entity/Domain.php (hunk 1 above)
Comment (andypost) [reply domain/src/Entity/Domain.php:479]: Probably that was in times befor interfaces for config and content got split

Comment (agentrickard) [reply domain/src/Entity/Domain.php:479]: I suspect you'll see a lot of that in the code cleanup.

**Comment by andypost** [comment]
I will care to cleanup and provide BC shims for loader&creator that's why I started to review & fix file by file

**Comment by agentrickard** [comment]
I'm a huge fan of creating convenience methods and functions for other developers, which is why we have things like DomainLoader::loadDefaultDomain().

Those might be something that can be moved to a more standard storage extension, but I would like to keep them.

**Comment by andypost** [comment]
Sure, it's a part of interface that's why BC should be kept

**Comment by agentrickard** [comment]
For all my ignorance of OOP, i was very diligent about creating interfaces. :-)

**Comment by andypost** [comment]
Yep, they are perfectly architectured!
So it's easy to navigate and polish implementation

**Comment by agentrickard** [comment]
I blame @crell for that. :-)

**Comment by Crell** [comment]
Blame accepted!

**Pull request merged**